Update llvm version used for tests.

helly25 opened the pull request:
Allow attribute llvm_versions to specify version requirements.

Update some (main and test) dependencies.
Mark omp targets as manual (likely to be normal in a follow up).
Allow llvm_version to be taken from environment (or bazel command line). This does for newer bazel versions respect hermeticity.
Simplify version handling and log the resolution only in one place.
Update default llvm target version to "latest:>=17.0.0,<20" (e.g. 19) but allow environment overrides.

helly25 committed:
Update llvm versions for tests to at least 17.0.0.
Allow attribute llvm_versions to accept version requirements.

helly25 committed:
Update llvm version used for tests.
Allow attribute llvm_versions to specify version requirements.

helly25 committed:
Further bump LLVM versions for tests.

helly25 committed:
Make some tests explicitly use llvm 16, as they do not yet work with llvm 17+.

helly25 committed:
Provide omp dependency via toolchain _name argument in run_tests.sh.

helly25 committed:
Must use two args to provide the define as on the intended machine (unlike mine) we end up getting quotes.

helly25 committed:
Merge branch 'bazel-contrib:master' into master

helly25 committed:
Merge branch 'bazel-contrib:master' into master

helly25 committed:
Ubuntu 20.04 does not have newer glibc as required by newer LLVM 16+. So we force version 15 for that. Possibly have to do so for some other tests as well.

helly25 committed:
Merge branch 'master' into update/test_llvm

helly25 committed:
Must be one newer...

helly25 committed:
Missing version piece

helly25 committed:
Merge remote-tracking branch 'origin' into pr/helly25/586

helly25 committed:
Allow to override the LLVM_VERSION via an environment variable for testing.

helly25 committed:
Make omp targets manual and cc back to 2.8

helly25 committed:
Downgrade.

helly25 committed:
Sync sys version back to 16.0.0

helly25 committed:
Oops, versions require a dict. This was not a version attribute.

helly25 commented:
@fmeum Please take another look.

fmeum commented on tests/BUILD.bazel:
```diff
@@ -101,6 +101,7 @@ cc_test(
     srcs = ["omp_test.c"],
     copts = ["-fopenmp"],
     linkopts = ["-fopenmp"],
+    tags = ["manual"],
```
Should we run these explicitly on some platforms?

helly25 replied:
Done: There are not many though since they do not work with new llvm versions. But at least they need to be explicitly disabled.

fmeum commented on toolchain/internal/llvm_distributions.bzl:
```diff
@@ -1159,24 +1168,57 @@ def _required_llvm_release_name(*, version_or_requirements, all_llvm_distributio
+        else:
+            llvm_version = env_default
+
+        #print("\nINFO: Read environment variable '{env_var}' = '{llvm_version}'".format(
```
Can we drop this?

helly25 committed:
Address comments.
Check whether all targets support omp again. If that fails we have to add some flags and other tag managment and determine why that is necessary.

helly25 committed:
Conditionally run omp_targets

helly25 committed:
Oversight: Missed adding the O flag in the last commit

helly25 committed:
Another manual tag is needed

helly25 committed:
Manual disabling omp tests and version handling improvements.

helly25 commented on tests/WORKSPACE:
```diff
@@ -134,9 +137,10 @@ llvm_toolchain(
 llvm_toolchain(
     name = "llvm_toolchain_with_system_llvm",
-    llvm_versions = LLVM_VERSIONS,
+    llvm_versions = {"": "16.0.0"},
```
Done.
